src/hydra/HydraAdmin.js

```diff
@@ -1,29 +1,48 @@
-import React, {Component} from 'react';
+import apiDocumentationParser from 'api-doc-parser/lib/hydra/parseHydraDocumentation';
 import PropTypes from 'prop-types';
-import parseHydraDocumentation from 'api-doc-parser/lib/hydra/parseHydraDocumentation';
+import React, {Component} from 'react';
 import AdminBuilder from '../AdminBuilder';
-import hydraClient from './hydraClient';
+import restClient from './hydraClient';
+
+class HydraAdmin extends Component {
+  static defaultProps = {
+    apiDocumentationParser,
+    restClient,
+  };
 
-export default class extends Component {
   static propTypes = {
-    entrypoint: PropTypes.string,
+    apiDocumentationParser: PropTypes.func,
+    entrypoint: PropTypes.string.isRequired,
+    restClient: PropTypes.func,
+  };
+
+  state = {
+    api: null,
   };
-  state = {api: null};
 
   componentDidMount() {
-    parseHydraDocumentation(this.props.entrypoint).then(api =>
-      this.setState({api: api}),
-    );
+    this.props
+      .apiDocumentationParser(this.props.entrypoint)
+      .then(api => {
+        console.log(api);
+        return api;
+      })
+      .then(api => this.setState({api}));
   }
 
   render() {
-    if (null === this.state.api) return <span>Loading...</span>;
-
-    let props = {...this.props};
-    if (!props.api) props.api = this.state.api;
-    if (!props.restClient)
-      props.restClient = hydraClient(this.props.entrypoint);
+    if (null === this.state.api) {
```

Comment:
Not related to this PR (note for the future): we should use a props here (like `loader`).